python/apps/taiga/src/taiga/users/services/__init__.py

```diff
@@ -234,7 +248,119 @@ async def update_user(user: User, full_name: str, lang: str) -> User:
+                select_related=["user", "role", "project", "project__workspace"],
+            )
+            if pj_membership and project_admin_role:
+                updated_pj_membership = await pj_memberships_repositories.update_project_membership(
+                    membership=pj_membership,
+                    values={"role": project_admin_role},
+                )
+                await pj_memberships_events.emit_event_when_project_membership_is_updated(
+                    membership=updated_pj_membership
+                )
+        else:
+            if project_admin_role:
+                created_pj_membership = await pj_memberships_repositories.create_project_membership(
+                    project=pj, role=project_admin_role, user=workspace_members[0]
+                )
+                await pj_memberships_events.emit_event_when_project_membership_is_created(
+                    membership=created_pj_membership
+                )
+
+    # delete ws memberships
+    ws_memberships = await ws_memberships_repositories.list_workspace_memberships(
+        filters={"user_id": user.id}, select_related=["user", "workspace"]
+    )
+    for ws_membership in ws_memberships:
+        deleted = await ws_memberships_repositories.delete_workspace_memberships(
+            filters={"id": ws_membership.id},
+        )
+        if deleted > 0:
+            await ws_memberships_events.emit_event_when_workspace_membership_is_deleted(membership=ws_membership)
+
+    # delete ws invitations
+    ws_invitations = await ws_invitations_repositories.list_workspace_invitations(
+        filters={"user": user},
+        select_related=["workspace"],
+    )
+    for ws_invitation in ws_invitations:
+        is_pending = True if ws_invitation.status == WorkspaceInvitationStatus.PENDING else False
+        deleted = await ws_invitations_repositories.delete_workspace_invitation(filters={"id": ws_invitation.id})
+        if deleted > 0 and is_pending:
+            await ws_invitations_events.emit_event_when_workspace_invitation_is_deleted(invitation=ws_invitation)
+
+    # delete pj memberships
+    pj_memberships = await pj_memberships_repositories.list_project_memberships(
+        filters={"user_id": user.id},
+        select_related=["user", "project", "project__workspace"],
+    )
+    for pj_membership in pj_memberships:
+        deleted = await pj_memberships_repositories.delete_project_membership(
+            filters={"id": pj_membership.id},
+        )
+        if deleted > 0:
+            await pj_memberships_events.emit_event_when_project_membership_is_deleted(membership=pj_membership)
+
+    # delete pj invitations
+    pj_invitations = await pj_invitations_repositories.list_project_invitations(
+        filters={"user": user},
+        select_related=["project"],
+    )
+    for pj_invitation in pj_invitations:
+        is_pending = True if pj_invitation.status == ProjectInvitationStatus.PENDING else False
```

Comment:
Why don't you add a new simple filter `"status": ProjectInvitationStatus.PENDING` to the call to `await pj_invitations_repositories.list_project_invitations`?
If I don't get it wrong you can save the for and conditional part.

Reply:
Because we have to delete all invitations but only pending has event.

Reply:
Yes, you're totally right. My apologies!